motionplan/cBiRRT.go

```diff
@@ -461,59 +458,69 @@ func (mp *cBiRRTMotionPlanner) smoothPath(
 	schan := make(chan node, 1)
 	defer close(schan)
 
-	for iter := 0; iter < toIter && len(inputSteps) > 4; iter++ {
-		select {
-		case <-ctx.Done():
-			mp.logger.Debug("CBiRRT timed out during smoothing, returning best path")
-			return inputSteps
-		default:
-		}
-		// Pick two random non-adjacent indices, excepting the ends
-
-		j := 2 + mp.randseed.Intn(len(inputSteps)-3)
-
-		i := mp.randseed.Intn(j) + 1
-
-		ok, hitCorners := smoothable(inputSteps, i, j, mp.corners)
-		if !ok {
-			continue
-		}
-
-		shortcutGoal := make(map[node]node)
-
-		iSol := inputSteps[i]
-		jSol := inputSteps[j]
-		shortcutGoal[jSol] = nil
-
-		// extend backwards for convenience later. Should work equally well in both directions
-		mp.constrainedExtend(ctx, mp.randseed, shortcutGoal, jSol, iSol, schan)
-		reached := <-schan
-
-		// Note this could technically replace paths with "longer" paths i.e. with more waypoints.
-		// However, smoothed paths are invariably more intuitive and smooth, and lend themselves to future shortening,
-		// so we allow elongation here.
-		dist := mp.planOpts.DistanceFunc(&Segment{StartConfiguration: inputSteps[i].Q(), EndConfiguration: reached.Q()})
-		if dist < mp.algOpts.JointSolveDist && len(reached.Q()) < j-i {
-			mp.corners[iSol] = true
-			mp.corners[jSol] = true
-			for _, hitCorner := range hitCorners {
-				mp.corners[hitCorner] = false
+	for numCornersToPass := 2; numCornersToPass > 0; numCornersToPass-- {
+		for iter := 0; iter < toIter/2 && len(inputSteps) > 4; iter++ {
```

Comment:
`len(inputSteps) > 4`

A path of len(3) could and should be attempted to be smoothed.